core/util/proxy.py

```diff
@@ -0,0 +1,115 @@
+"""
+OWASP Maryam!
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+#import self.framework.requests
+from bs4 import BeautifulSoup
+import time
+#Python proxy module. Contributed by: @anuran-roy under @saeeddhqan
+class main:
+
+    def __init__(self, verbose = False):
+        '''
+        use : Determine whether to use proxy or not.
+        '''
+        self.use=self._global_options_['autoproxy']
+        self.verbose = verbose
+        self.iplist = []
+
+    def getproxy(self):
+        if self.use == True:
+            self.framework.output("[PROXY] Gathering proxies. It's quite time consuming, so do some pushups in the meantime :P")
+            URL = [ 'https://premproxy.com/list/' ]
+            i = 2
+            while i<=8:
+                purl = 'https://premproxy.com/list/0'+str(i)+".htm"
+                URL.append(purl)
+                i += 1    
+            ips = []
+            ips2 = []
+            purl = URL[0]
+            try:
+                for purl in URL:
+                    r = self.framework.requests.get(purl)
+                    sp = BeautifulSoup(r.content,'html.parser')
+                    lst = list(sp.find_all('input',attrs={"name" : "proxyIp[]"}))
+                    for l in lst:
+                        ip = str(l.get('value'))[0:-6]
+                        ips.append(ip)
+                    time.sleep(2)
+                #self.framework.output(ips[0:5])
+                self.framework.output("[PROXY] Proxies gathered.")
+            except Exception as e1:
+                self.framework.error(f"[PROXY] An error occured. Error details: {e1}")    
+            j = 0
+            while(j<len(ips)):
+                ips[j] = f"http://{ips[j]}"
+                j += 1
+            j = 0
+            self.framework.output("[PROXY] Now checking for availability of the proxies...")
+            while(j<len(ips)):
+                try:
```

Comment:
hard-coding

Reply:
I reused the variable. I guess that'd save a few extra bytes of memory from being used. 😅